### PR TITLE
Use ruff for our python lints

### PR DESCRIPTION
**Pull request by jleibs**
Ruff seems likely to be faster and more stable, though it did want to re-sort a lot of our imports.
These mostly seem to make sense, since from the perspective of examples, `rerun` is just another external dep and should be sorted as such.

```
$ time just py-lint
black --check --config rerun_py/pyproject.toml --diff examples rerun_py scripts
All done! ✨ 🍰 ✨
80 files would be left unchanged.
blackdoc --check examples rerun_py scripts
All done! ✨ 🍰 ✨
82 files would be left unchanged.
ruff check --config rerun_py/pyproject.toml  examples rerun_py scripts

real    0m0.384s
user    0m1.448s
sys     0m0.236s
```

And looks to run consistently in CI:
image

https://github.com/rerun-io/rerun/issues/1377

### Checklist
* [x] I have read and agree to [Contributor Guide](../CONTRIBUTING.md) and the [Code of Conduct](../CODE_OF_CONDUCT.md)
* [ ] I've included a screenshot or gif (if applicable)
* [ ] I've added a line to `CHANGELOG.md` (if this is a big enough change to warrant it)

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -49,20 +49,18 @@ py-build:
...
     pyupgrade --py37-plus `find rerun_py/rerun/ -name "*.py" -type f`
+    ruff --fix --config rerun_py/pyproject.toml  {{py_folders}}
 
 # Check that all the requirements.txt files for all the examples are correct
 py-requirements:
     find examples/python/ -name main.py | xargs -I _ sh -c 'cd $(dirname _) && echo $(pwd) && pip-missing-reqs . || exit 255'
 
 # Run linting
-py-lint: py-requirements
+py-lint:
```
Comment (jleibs) [inline justfile:60]: The dep on py-requirements still made this quite slow, so I pulled it out and added it to CI.

Comment (emilk) [reply justfile:60]: Good call!

File: justfile [hunk 2]
```diff
@@ -49,20 +49,18 @@ py-build:
...
     black --check --config rerun_py/pyproject.toml --diff {{py_folders}}
     blackdoc --check {{py_folders}}
-    isort --check {{py_folders}}
-    mypy --no-warn-unused-ignore
```
Comment (emilk) [inline justfile:64]: does `ruff` replace `mypy` too? Does it really do type-checking?

File: rerun_py/pyproject.toml [hunk 3]
```diff
@@ -39,6 +39,38 @@ rerun = "rerun.__main__:main"
...
+    "D416",
+]
+line-length = 120
+select = [
+    "D",
+    "E",
+    "F",
+    "I",
+    "TID",
+    "W",
```
Comment (emilk) [inline rerun_py/pyproject.toml:68]: what is this? O.o

File: rerun_py/pyproject.toml [hunk 4]
```diff
@@ -39,6 +39,38 @@ rerun = "rerun.__main__:main"
...
+ignore = [
+    "D1",
+    "D107",
+    "D202",
+    "D203",
+    "D212",
+    "D401",
+    "D402",
+    "D415",
+    "D416",
```
Comment (emilk) [inline rerun_py/pyproject.toml:59]: In `.flake8` we have documented why each of these are ignored

File: rerun_py/pyproject.toml [hunk 5]
```diff
@@ -39,6 +39,38 @@ rerun = "rerun.__main__:main"
 line-length = 120
 target-version = ["py38"]
 
+[tool.ruff]
+extend-exclude = [
+    "venv/",
+    "target/",
+    "examples/python/objectron/proto/objectron/proto.py",
+    "examples/python/stable_diffusion/huggingface_pipeline.py",
+    "examples/python/colmap/read_write_model.py",
```
Comment (emilk) [inline rerun_py/pyproject.toml:48]: In `.flake8` we have documented why these are excluded

**Review by emilk — COMMENTED** [review]
Let's delete `.flake8` too

File: justfile (hunk 2 above)
Comment (jleibs) [reply justfile:64]: Sadly no -- I mentally mixed up mypy and flake8 -- will restore that. As much as I want to delete it.

Comment (jleibs) [reply justfile:64]: Restored and updated to version 1.0.1

File: rerun_py/pyproject.toml (hunk 3 above)
Comment (jleibs) [reply rerun_py/pyproject.toml]: Added comments.

File: .vscode/extensions.json [hunk 6]
```diff
@@ -2,6 +2,7 @@
     // See https://go.microsoft.com/fwlink/?LinkId=827846
     // for the documentation about the extensions.json format
     "recommendations": [
+        "charliermarsh.ruff", // Ruff for linting
```
Comment (emilk) [inline .vscode/extensions.json:5]: niiiice

**Review by emilk — APPROVED** [review]
so nice!

**Pull request merged**